aten/src/ATen/native/Indexing.cpp

```diff
@@ -251,6 +251,10 @@ Tensor & _index_put_impl_(Tensor & self, TensorList indices, const Tensor & valu
     AT_INDEX_ERROR("too many indices for tensor of dimension ", self.dim(), " (got ", indices.size(), ")");
   }
   if (accumulate && self.device().type() == kCUDA) {
+      if (value.device() != self.device()) {
+        AT_ERROR("expected device ", self.device(), " but got device ",
```

Comment:
Isn't AT_ERROR deprecated in favor of TORCH_CHECK(false, ...)?

Reply:
You are right. There are 336 instances of AT_ERROR in ATen though.